pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -53,6 +53,10 @@ type SpdyRoundTripper struct {
 
 	// Dialer is the dialer used to connect.  Used if non-nil.
 	Dialer *net.Dialer
+
+	// proxier knows which proxy to use given a request, defaults to http.ProxyFromEnvironment
+	// Used primarily for mocking the proxy discovery in tests.
```

Comment:
rather than defaulting in `NewSpdyRoundTripper`, I would default in `dial` if s.proxier is nil. That ensures we don't break anyone using the struct directly, and matches the comment better